controller/server/main.go

```diff
@@ -138,6 +137,59 @@ func validatePath(path string) (string, error) {
 	return path, nil
 }
 
+func (s *server) CreateTopology(ctx context.Context, req *cpb.CreateTopologyRequest) (*cpb.CreateTopologyResponse, error) {
+	log.Infof("Received CreateTopology request: %v", req)
+	bp, err := validatePath(req.Topology.Name)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid topology path: %v", err)
+	}
+	p := topo.TopologyParams{
+		TopoName:       req.Topology.Name,
+		TopoNewOptions: []topo.Option{topo.WithBasePath(filepath.Dir(bp))},
+		Kubecfg:        req.Kubecfg,
+	}
+	err = topo.CreateTopology(ctx, p)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create topology: %v", err)
+	}
+
+	return &cpb.CreateTopologyResponse{
+		TopologyName: req.Topology.GetName(),
+		State:        cpb.TopologyState_TOPOLOGY_STATE_RUNNING,
```

Comment:
should this state be stored? i.e. is thera some other get call which would return this status?

Reply:
The latest state can be got via ShowTopology() call.